### PR TITLE
fix(codewhisperer): noisy logs every time UI is updated

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Solution

https://github.com/aws/aws-toolkit-vscode/assets/96078566/33ada40b-75e1-4284-a3cc-1404d2bf47b6

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/util/authUtil.ts [hunk 1]
```diff
@@ -283,9 +283,9 @@ export class AuthUtil {
         return this.conn.getCredentials()
     }
 
-    public isConnectionValid(): boolean {
+    public isConnectionValid(silent: boolean = false): boolean {
         const connectionValid = this.conn !== undefined && !this.secondaryAuth.isConnectionExpired
-        if (connectionValid === false) {
+        if (!silent) {
```
Comment (Will-ShaoHua) [inline packages/core/src/codewhisperer/util/authUtil.ts:288]: actually I prefer to remove this logging here, but want to align with folks here

**Comment by justinmk3** [comment]
The docs explain how to format commit messages / PR titles: https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#commit-messages

> Prefix the subject with `type(topic)`: ([conventional commits](https://www.conventionalcommits.org/) format): this again helps humans (and scripts) scan and omit ranges of the history at a glance.

`(fix codewhisperer)` should be `fix(codewhisperer)`

File: packages/core/src/codewhisperer/util/authUtil.ts (hunk 1 above)
Comment (justinmk3) [reply packages/core/src/codewhisperer/util/authUtil.ts:288]: You can use `onceChanged` to only log if the previous value is different: https://github.com/aws/aws-toolkit-vscode/blob/2c4ea6516c5249c048b301d5826f4b0cfe16bf0e/packages/core/src/shared/utilities/functionUtils.ts#L54

Comment (justinmk3) [reply packages/core/src/codewhisperer/util/authUtil.ts:288]: Example: https://github.com/aws/aws-toolkit-vscode/blob/e5fdf4109207f801a9b91a4d162ef1fb8d1bffd1/packages/core/src/shared/sam/cli/samCliSettings.ts#L60

Comment (justinmk3) [reply packages/core/src/codewhisperer/util/authUtil.ts:288]: You can use `onceChanged` to only log if the previous value is different: https://github.com/aws/aws-toolkit-vscode/blob/e5fdf4109207f801a9b91a4d162ef1fb8d1bffd1/packages/core/src/shared/sam/cli/samCliSettings.ts#L60

Comment (Will-ShaoHua) [reply packages/core/src/codewhisperer/util/authUtil.ts:288]: gotcha, that would be a better solution here, thanks Justin

**Comment by justinmk3** [comment]
The CI logs show other noisy logs, can they get the same fix?:

```
    secondaryAuth connection id = f4d24020-97e3-443f-96e2-a927a1af0c44
    secondaryAuth connection status = valid
    secondaryAuth connection id = f4d24020-97e3-443f-96e2-a927a1af0c44
    secondaryAuth connection status = valid
    secondaryAuth connection id = f4d24020-97e3-443f-96e2-a927a1af0c44
    secondaryAuth connection status = valid
    secondaryAuth connection id = f4d24020-97e3-443f-96e2-a927a1af0c44
    secondaryAuth connection status = valid
    secondaryAuth connection id = f4d24020-97e3-443f-96e2-a927a1af0c44
    secondaryAuth connection status = valid
    secondaryAuth connection id = f4d24020-97e3-443f-96e2-a927a1af0c44
    secondaryAuth connection status = valid
    secondaryAuth connection id = f4d24020-97e3-443f-96e2-a927a1af0c44
```

File: packages/core/src/codewhisperer/util/authUtil.ts [hunk 2]
```diff
@@ -299,17 +299,33 @@ export class AuthUtil {
...
-                `codewhisperer: isValidCodeWhispererConnection = ${isValidCodeWhispererCoreConnection(this.conn)}`
-            )
-        }
+
+        this.logCnnection()
+
         return connectionExpired
     }
 
+    private logCnnection() {
```
Comment (justinmk3) [inline packages/core/src/codewhisperer/util/authUtil.ts:308]:
```suggestion
    private logConnection() {
```

File: packages/core/src/codewhisperer/util/authUtil.ts [hunk 3]
```diff
@@ -299,17 +299,33 @@ export class AuthUtil {
...
+        const connectionValid = this.conn !== undefined && !this.secondaryAuth.isConnectionExpired
+        const connectionExpired =
+            this.secondaryAuth.isConnectionExpired &&
+            this.conn !== undefined &&
+            isValidCodeWhispererCoreConnection(this.conn)
```
Comment (justinmk3) [inline packages/core/src/codewhisperer/util/authUtil.ts:313]: could avoid repeating this logic if `isConnectionExpired` and `isConnectionValid` took a `log:boolean` parameter:
```suggestion
        const connectionValid = isConnectionValid(false)
        const connectionExpired = isConnectionExpired(false)
```

Comment (Will-ShaoHua) [reply packages/core/src/codewhisperer/util/authUtil.ts:313]: gotcha

File: packages/toolkit/package.json [hunk 4]
```diff
@@ -52,8 +52,4185 @@
     ],
     "main": "./dist/src/main",
     "browser": "./dist/src/mainWeb",
-    "engines": "This field will be autopopulated from the core module during debugging and packaging.",
```
Comment (Will-ShaoHua) [inline packages/toolkit/package.json:55]: will revert

**Review by justinmk3 — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

**Pull request merged**